joomla/joomla-cms
# Better error message for MySQL 8 workaround

### Summary of Changes
Joomla 4.0 don't work with MySQL 8 without a workaround.

Please follow these introductions from this [wiki](https://docs.joomla.org/Joomla_and_MySQL_8#Workaround_to_get_Joomla_working_with_MySQL_8) I wrote to make Joomla 4.0 and MySQL 8 working.
Joomla 3.8, 3.9 are working fine as well with this workaround.

I add a link to the error message to help some users who wants to use Joomla with MySQL 8.

Before

<img width="676" alt="bildschirmfoto 2018-07-30 um 09 18 08" src="https://user-images.githubusercontent.com/8920003/43515703-825f9440-9583-11e8-8d89-a1e0d96e7b69.png">

After

<img width="677" alt="bildschirmfoto 2018-08-01 um 12 18 31" src="https://user-images.githubusercontent.com/8920003/43516268-5c82e7ac-9585-11e8-970e-e0c5d46fe7ce.png">

### Test instructions
Please open your configuration file <tt>sudo nano /etc/my.cnf</tt> (Please note that your file may be under a different directory) and add the following configuration:</translate> 
```sql
[mysqld]
default-authentication-plugin=mysql_native_password
```
<translate> If you don't have access to your config file then you can update your user as follows:</translate>
```sql
ALTER USER 'username'@'localhost' IDENTIFIED WITH mysql_native_password BY 'password';
```
<translate> Replace username with the name of the user account and password with the password belonging to the account. Restart MySQL and you are done, well only if you have Joomla 3.8 or 3.9 installed.</translate>

<translate> If you want to work with Joomla 4.0 you have to do some extra work, because the administration dashboard is blank after the installation with Joomla 4.0 and MySQL 8.</translate>

<translate>
Fortunately there is a fix for that too! 
MySQL 8 is returning <tt>Incorrect date value: '0000-00-00'</tt> after running following insertion from the installation file <tt>/installation/sql/mysql/joomla.sql</tt>.</translate> 

<translate> The following query must be run to fix this. Replace #_ with your table prefix.</translate>
```sql
UPDATE `#__modules` SET `checked_out_time` = '1000-01-01 00:00:00', `publish_up` = '1000-01-01 00:00:00', `publish_down` = '1000-01-01 00:00:00';
```

<translate> This is happening because since MySQL 5.7, MySQL stops supporting zeros value in date / datetime. Restart MySQL and you are done!</translate>

<translate>
